### PR TITLE
Quake: init 0.3.0

### PR DESCRIPTION
**Pull request by Eliot00**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/quake/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  cargoSha256 = "sha256:1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
+    description = "A knowledge management meta-framework for geeks.";
```
Comment (legendofmiracles) [inline pkgs/applications/misc/quake/default.nix:21]:
```suggestion
    description = "A knowledge management meta-framework for geeks";
```
Descriptions never have full-stops at the end

**Review by legendofmiracles — COMMENTED** [review]
Your PR shouldn't target `nixos:release-21.11` but rather `master`. If you want it in 21.11 we can backport it

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1863,6 +1863,8 @@ with pkgs;
 
   passExtensions = recurseIntoAttrs pass.extensions;
 
+  quake = callPackage ../applications/misc/quake { };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:1866]:
```suggestion
  phodal-quake = callPackage ../applications/misc/phodal-quake { };
```

Comment (Eliot00) [reply pkgs/top-level/all-packages.nix:1866]: The quake repo will move to this [public org](https://github.com/inherd), can I just use the name `quake`?

Comment (veprbl) [reply pkgs/top-level/all-packages.nix:1866]: It's not forbidden, but can cause a confusion https://repology.org/project/quake/versions

Comment (Eliot00) [reply pkgs/top-level/all-packages.nix:1866]: Thanks for reply, for avoid ambiguity, I will use org's name as prefix, `inherd-quake`

File: pkgs/applications/misc/inherd-quake/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "quake";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "phodal";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:1f7k68g18g3dpnrsmhgmz753bly1i3f4lmsljiyp9ap0c6w8ahgg";
```
Comment (bobby285271) [inline pkgs/applications/misc/inherd-quake/default.nix:18]: Nit *(This is just my personal preference)*

```suggestion
    sha256 = "1f7k68g18g3dpnrsmhgmz753bly1i3f4lmsljiyp9ap0c6w8ahgg";
```

File: pkgs/applications/misc/inherd-quake/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  cargoSha256 = "sha256:1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
```
Comment (bobby285271) [inline pkgs/applications/misc/inherd-quake/default.nix:21]:
```suggestion
  cargoSha256 = "1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -1915,6 +1915,8 @@ with pkgs;
 
   passExtensions = recurseIntoAttrs pass.extensions;
 
+  inherd-quake = callPackage ../applications/misc/inherd-quake { };
```
Comment (bobby285271) [inline pkgs/top-level/all-packages.nix:1918]: We commonly put this in top-level.

```suggestion
  inherd-quake = callPackage ../applications/misc/inherd-quake {
    inherit (darwin.apple_sdk.frameworks) CoreServices Security;
  };
```

File: pkgs/applications/misc/inherd-quake/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+, stdenv
+, darwin
```
Comment (bobby285271) [inline pkgs/applications/misc/inherd-quake/default.nix:7]:
```suggestion
, CoreServices
, Security
```

File: pkgs/applications/misc/inherd-quake/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  cargoSha256 = "sha256:1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ] ++ (
+    lib.optional stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.CoreServices
+    ]
+  );
```
Comment (bobby285271) [inline pkgs/applications/misc/inherd-quake/default.nix:30]: Maybe use <code>lib.optional<b>s</b></code>?

```suggestion
  buildInputs = [
    openssl
  ] ++ lib.optionals stdenv.isDarwin [
    CoreServices
    Security
  ];
```

**Comment by bobby285271** [comment]
Seems that ofborg is on holiday at the moment...

@ofborg build inherd-quake

**Comment by bobby285271** [comment]
@ofborg build inherd-quake

**Review by bobby285271 — APPROVED** [review]
Diff looks good to me, builds and prints help menu.

**Pull request merged**